### PR TITLE
add a slot-duration getter to babe config

### PR DESCRIPTION
**Pull request by rphmeier**
Will be used in approval-voting in Polkadot

### CODE REVIEW

File: client/consensus/babe/src/lib.rs [hunk 1]
```diff
@@ -342,6 +342,11 @@ impl Config {
 			}
 		}
 	}
+
+	/// Get the inner slot duration, in milliseconds.
+	pub fn slot_duration(&self) -> u64 {
+		self.0.slot_duration()
```
Comment (bkchr) [inline client/consensus/babe/src/lib.rs:348]:
```suggestion
	pub fn slot_duration(&self) -> sc_consensus_slots::SlotDuration<BabeGenesisConfiguration> {
		self.0
```

**Review by bkchr — APPROVED** [review]
I would return the `SlotDuration` to directly use the new type wrapper.

**Comment by rphmeier** [comment]
@bkchr Why? That's an implementation detail of the fact that BABE uses `sc_consensus_slots` internally.

**Comment by bkchr** [comment]
You add a getter for slot duration and we have some type that represents this slot duration. I don't see this as implementation detail. There are also maybe other use cases where you directly want the slot duration as type again because you call something in the sc_consensus. Every other use case will also not be that complicated as the slot duration type exposes the internal data.

**Comment by rphmeier** [comment]
`sc_consensus_slots` is an implementation detail. Why should we leak it into the public API of `sc-consensus-babe`?

**Comment by bkchr** [comment]
I will not insist on this. Fine to ignore my comment.

**Review by andresilva — APPROVED** [review]

**Comment by andresilva** [comment]
bot merge

**Pull request merged**